nifi-external/nifi-kafka-connect/nifi-kafka-connector/src/main/java/org/apache/nifi/kafka/connect/StatelessKafkaConnectorUtil.java

```diff
@@ -269,4 +275,53 @@ private static File detectNarDirectory() {
         logger.info("Detected NAR Directory to be {}", narDirectory.getAbsolutePath());
         return narDirectory;
     }
+
+    private static void checkWorkingDirectoryIntegrity(final File workingDirectory) {
+        purgeIncompleteUnpackedNars(new File(new File(workingDirectory, "nar"), "extensions"));
```

Comment:
It looks like purge method could be abstracted to a separate utility, which would make it much easier to unit test.